dcrec/secp256k1/ellipticadaptor_test.go

```diff
@@ -1,65 +1,421 @@
+		y2:   "131c670d414c4546b88ac3ff664611b1c38ceb1c21d76369d7a7a0969d61d97d",
+		x3:   "d74bf844b0862475103d96a611cf2d898447e288d34b360bc885cb8ce7c00575",
+		y3:   "131c670d414c4546b88ac3ff664611b1c38ceb1c21d76369d7a7a0969d61d97d",
+	}, {
+		// Addition with the point at infinity (right hand side).
+		name: "P + ∞ = P",
+		x1:   "d74bf844b0862475103d96a611cf2d898447e288d34b360bc885cb8ce7c00575",
+		y1:   "131c670d414c4546b88ac3ff664611b1c38ceb1c21d76369d7a7a0969d61d97d",
+		x2:   "0",
+		y2:   "0",
+		x3:   "d74bf844b0862475103d96a611cf2d898447e288d34b360bc885cb8ce7c00575",
+		y3:   "131c670d414c4546b88ac3ff664611b1c38ceb1c21d76369d7a7a0969d61d97d",
+	}, {
+		// Addition with different x values.
+		name: "P(x1, y1) + P(x2, y2)",
+		x1:   "34f9460f0e4f08393d192b3c5133a6ba099aa0ad9fd54ebccfacdfa239ff49c6",
+		y1:   "0b71ea9bd730fd8923f6d25a7a91e7dd7728a960686cb5a901bb419e0f2ca232",
+		x2:   "d74bf844b0862475103d96a611cf2d898447e288d34b360bc885cb8ce7c00575",
+		y2:   "131c670d414c4546b88ac3ff664611b1c38ceb1c21d76369d7a7a0969d61d97d",
+		x3:   "fd5b88c21d3143518d522cd2796f3d726793c88b3e05636bc829448e053fed69",
+		y3:   "21cf4f6a5be5ff6380234c50424a970b1f7e718f5eb58f68198c108d642a137f",
+	}, {
+		// Addition with same x opposite y.
+		name: "P(x, y) + P(x, -y) = ∞",
+		x1:   "34f9460f0e4f08393d192b3c5133a6ba099aa0ad9fd54ebccfacdfa239ff49c6",
+		y1:   "0b71ea9bd730fd8923f6d25a7a91e7dd7728a960686cb5a901bb419e0f2ca232",
+		x2:   "34f9460f0e4f08393d192b3c5133a6ba099aa0ad9fd54ebccfacdfa239ff49c6",
+		y2:   "f48e156428cf0276dc092da5856e182288d7569f97934a56fe44be60f0d359fd",
+		x3:   "0",
+		y3:   "0",
+	}, {
+		// Addition with same point.
+		name: "P(x, y) + P(x, y) = 2P",
+		x1:   "34f9460f0e4f08393d192b3c5133a6ba099aa0ad9fd54ebccfacdfa239ff49c6",
+		y1:   "0b71ea9bd730fd8923f6d25a7a91e7dd7728a960686cb5a901bb419e0f2ca232",
+		x2:   "34f9460f0e4f08393d192b3c5133a6ba099aa0ad9fd54ebccfacdfa239ff49c6",
+		y2:   "0b71ea9bd730fd8923f6d25a7a91e7dd7728a960686cb5a901bb419e0f2ca232",
+		x3:   "59477d88ae64a104dbb8d31ec4ce2d91b2fe50fa628fb6a064e22582196b365b",
+		y3:   "938dc8c0f13d1e75c987cb1a220501bd614b0d3dd9eb5c639847e1240216e3b6",
+	}}
+
+	curve := S256()
+	for _, test := range tests {
+		// Parse the test data.
+		x1, y1 := fromHex(test.x1), fromHex(test.y1)
+		x2, y2 := fromHex(test.x2), fromHex(test.y2)
+		x3, y3 := fromHex(test.x3), fromHex(test.y3)
+
+		// Ensure the test data is using points that are actually on the curve
+		// (or the point at infinity).
+		if !isValidAffinePoint(x1, y1) {
+			t.Errorf("%s: first point is not on curve", test.name)
+			continue
+		}
+		if !isValidAffinePoint(x2, y2) {
+			t.Errorf("%s: second point is not on curve", test.name)
+			continue
+		}
+		if !isValidAffinePoint(x3, y3) {
+			t.Errorf("%s: expected point is not on curve", test.name)
+			continue
+		}
+
+		// Add the two points and ensure the result matches expected.
+		rx, ry := curve.Add(x1, y1, x2, y2)
+		if rx.Cmp(x3) != 00 || ry.Cmp(y3) != 0 {
```

Comment:
`00` could just be `0`? Unsure if this has a special meaning.

Reply:
Indeed.  Updated.